Runtimes/Core/CMakeLists.txt

```diff
@@ -47,6 +48,13 @@ set(SwiftCore_SWIFTC_SOURCE_DIR
   "${PROJECT_SOURCE_DIR}/../../"
   CACHE FILEPATH "Path to the root source directory of the Swift compiler")
 
+# Hook point for vendor-specific private extensions to the build system
+# Allowed extension points:
+#   - DefaultSettings.cmake
+#   - Settings.cmake
+set(SwiftCore_PRIVATE_MODULE_DIR "${SwiftCore_CMAKE_MODULES_DIR}/private"
```

Comment:
What do you think of `SwiftCore_VENDOR_MODULE_DIR`? This is the vendor specific settings and makes it clear when you concatenate them even if you don't know what it is (e.g. `${SwiftCore_VENDOR_MODULE_DIR}/DefaultSettings.cmake` vs `${SwiftCore_PRIVATE_MODULE_DIR}/DefaultSettings.cmake`).

Reply:
Sure, sounds good to me